Formula/gnupg.rb

```diff
@@ -13,70 +13,92 @@ class Gnupg < Formula
+    args << "--with-readline=#{Formula["readline"].opt_prefix}" if build.with? "readline"
 
     system "./configure", *args
-    system "make"
-    system "make", "check"
 
-    # we need to create these directories because the install target has the
-    # dependency order wrong
-    [bin, libexec/"gnupg"].each(&:mkpath)
     system "make", "install"
 
-    # https://lists.gnupg.org/pipermail/gnupg-devel/2016-August/031533.html
-    inreplace bin/"gpg-zip1", "GPG=gpg", "GPG=gpg1"
+    # "make check" cannot run before "make install"
+    # Reported 1 Mar 2017 https://bugs.gnupg.org/gnupg/issue2979
+    system "make", "check"
 
-    # Although gpg2 support should be pretty universal these days
-    # keep vanilla `gpg` executables available, at least for now.
-    %w[gpg-zip1 gpg1 gpgsplit1 gpgv1].each do |cmd|
-      (libexec/"gpgbin").install_symlink bin/cmd => cmd.to_s.sub(/1/, "")
-    end
+    bin.install "tools/gpgsplit" => "gpgsplit2" if build.with? "gpgsplit"
   end
 
-  def caveats; <<-EOS.undent
-    This formula does not install either `gpg` or `gpgv` executables into
-    the PATH.
+  def post_install
+    (var/"run").mkpath
+  end
 
-    If you simply require `gpg` and `gpgv` executables without explicitly
-    needing GnuPG 1.x we recommend:
-      brew install gnupg2
+  def caveats; <<-EOS.undent
+    Once you run the new gpg2 binary you will find it incredibly
+    difficult to go back to using `gnupg@1.x` or `gnupg@2.0`
+    The new 2.1.x moves to a new keychain format that can't be
+    and won't be understood by the 2.0.x branch or lower.
 
-    If you really need to use these tools without the "1" suffix you can
-    add a "gpgbin" directory to your PATH from your #{shell_profile} like:
+    If you use this formula for a while and decide
+    you don't like it, you will lose the keys you've imported since.
+    For this reason, we strongly advise that you make a backup
+    of your `~/.gnupg` directory.
 
-        PATH="#{opt_libexec}/gpgbin:$PATH"
+    For full details of the changes, please visit:
+      https://www.gnupg.org/faq/whats-new-in-2.1.html
 
-    Note that doing so may interfere with GPG-using formulae installed via
-    Homebrew.
+    If you are upgrading to gnupg 2.1.x from gnupg 2.0.x you should execute:
+      `killall gpg-agent && gpg-agent --daemon`
+    After install. See:
+      https://github.com/Homebrew/homebrew-versions/issues/681
     EOS
   end
 
   test do
-    (testpath/"batchgpg").write <<-EOS.undent
-      Key-Type: RSA
-      Key-Length: 2048
-      Subkey-Type: RSA
-      Subkey-Length: 2048
-      Name-Real: Testing
-      Name-Email: testing@foo.bar
-      Expire-Date: 1d
-      %commit
-    EOS
-    system bin/"gpg1", "--batch", "--gen-key", "batchgpg"
-    (testpath/"test.txt").write "Hello World!"
-    system bin/"gpg1", "--armor", "--sign", "test.txt"
-    system bin/"gpg1", "--verify", "test.txt.asc"
+    system bin/"gpgconf"
```

Comment:
It should be safe to keep the stronger existing test from the original formula here given Homebrew captures `$HOME` for testing.